integrations/tensorflow/e2e/concat_test.py

```diff
@@ -51,40 +52,49 @@ def concat2axis(self, a, b):
 
 
 @tf_test_utils.compile_module(ConcatOpsModule)
-class ConcatOpsTest(tf_test_utils.CompiledModuleTestCase):
+class ConcatOpsTest(tf_test_utils.TracedModuleTestCase):
 
   def test_concat_zero_dim(self):
-    tf_utils.set_random_seed()
-    m = self.get_module()
-    a = tf.random.uniform([1, 5, 0], dtype=tf.float32)
-    b = tf.random.uniform([1, 5, 1], dtype=tf.float32)
-    dst = m.concat_zero_dim(a, b)
-    dst.assert_all_close()
-
-  def concat0axis(self):
-    tf_utils.set_random_seed()
-    m = self.get_module()
-    a = tf.random.uniform([1, 5, 1], dtype=tf.float32)
-    b = tf.random.uniform([1, 5, 1], dtype=tf.float32)
-    dst = m.concat_zero_dim(a, b)
-    dst.assert_all_close()
-
-  def concat1axis(self):
-    tf_utils.set_random_seed()
-    m = self.get_module()
-    a = tf.random.uniform([1, 5, 1], dtype=tf.float32)
-    b = tf.random.uniform([1, 5, 1], dtype=tf.float32)
-    dst = m.concat_zero_dim(a, b)
-    dst.assert_all_close()
-
-  def concat2axis(self):
-    tf_utils.set_random_seed()
-    m = self.get_module()
-    a = tf.random.uniform([1, 5, 1], dtype=tf.float32)
-    b = tf.random.uniform([1, 5, 1], dtype=tf.float32)
-    dst = m.concat_zero_dim(a, b)
-    dst.assert_all_close()
 
+    def concat_zero_dim(module):
+      tf_utils.set_random_seed()
+      a = tf_utils.uniform([1, 5, 0])
+      b = tf_utils.uniform([1, 5, 1])
+      module.concat_zero_dim(a, b)
+
+    self.compare_backends(concat_zero_dim)
+
+  def test_concat0axis(self):
+
+    def concat0axis(module):
+      tf_utils.set_random_seed()
+      a = tf_utils.uniform([1, 5, 1])
+      b = tf_utils.uniform([1, 5, 1])
+      module.concat0axis(a, b)
+
+    self.compare_backends(concat0axis)
+
+
+  def test_concat1axis(self):
+
+    def concat1axis(module):
+      tf_utils.set_random_seed()
```

Comment:
should we set random seed automatically inside compare_backends?

Reply:
Good idea! Do we have any tests that use modules with randomness in their method calls? Thinking ahead about how that would interact with testing during a benchmark warmup. If/when we do, should we have a default seed (e.g. `0`) that we always use before creating or recreating a trace?

Reply:
I don't think we even have a design yet for iree compiled modules producing random numbers / how they get seeded in a way we can reliably test. I think we can ignore the issue for now.

Reply:
Updated the README to reflect this and added some information about writing tests.